File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

enum
{
    E_A,
    E_B
};


bool cmp(pair<int,int>& a, pair<int,int>& b)
{
    if(a.first == b.first)
    {
        return a.second > b.second;
    }

    return a.first < b.first;
}

int solution(vector<int> A, vector<int> B) {
    int answer = 0;
    
    vector<pair<int, int>> total;
    
    for(int i = 0 ; i < A.size(); i++)
    {
        total.push_back({A[i], E_A});
        total.push_back({B[i], E_B});
    }
    
    sort(total.begin(), total.end(), cmp);
    
    int Acnt = 0;
    
    for(auto t : total)
    {
        if(t.second == E_B)
        {
            if(Acnt > 0)
            {
                answer++;
                Acnt--;
            }
        }
        else Acnt++;
    }
    
    return answer;
}

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int solution(int[] A, int[] B) {
        int answer = 0;
        
        var nums = new List<(int, int)>();
        
        foreach(var a in A)
        {
            nums.Add((a, 1));
        }
        
        foreach(var b in B)
        {
            nums.Add((b, 0));
        }
        
        nums.Sort();
        
        int Acnt = 0;
        for(int i = 0 ; i < nums.Count; i++)
        {
            var num = nums[i].Item1;
            var type = nums[i].Item2;
            
            if(type == 1) Acnt++;
            else 
            {
                if(Acnt > 0)
                {
                    answer++;
                    Acnt--;
                }
            }
        }
        
        return answer;
    }
}